src/app/data-planes/sources.ts

```diff
@@ -47,70 +20,53 @@ export type MeshGatewayDataplaneSource = DataSourceResponse<MeshGatewayDataplane
+  return arr.includes(item as T[number])
+}
 
-      return api.getDataplaneData({ mesh, dppName: name, dataPath })
+export const sources = (api: KumaApi) => {
+  return defineSources({
+    '/meshes/:mesh/dataplanes/:name': async (params) => {
+      return api.getDataplaneFromMesh(params)
     },
 
-    '/meshes/:mesh/dataplanes/:name/sidecar-dataplane-policies': (params: DetailParams, source: Closeable) => {
-      source.close()
-
+    '/meshes/:mesh/dataplanes/:name/data-path/:dataPath': async (params) => {
       const { mesh, name } = params
+      const dataPath = includes(['xds', 'clusters', 'stats'] as const, params.dataPath) ? params.dataPath : 'xds'
 
-      return api.getSidecarDataplanePolicies({ mesh, name })
+      return api.getDataplaneData({
+        mesh,
+        dppName: name,
+        dataPath,
+      })
     },
 
-    '/meshes/:mesh/dataplanes/:name/rules': (params: DetailParams, source: Closeable) => {
-      source.close()
-
-      const { mesh, name } = params
-
-      return api.getDataplaneRules({ mesh, name })
+    '/meshes/:mesh/dataplanes/:name/sidecar-dataplane-policies': async (params) => {
+      return api.getSidecarDataplanePolicies(params)
     },
 
-    '/meshes/:mesh/dataplanes/:name/gateway-dataplane-policies': (params: DetailParams, source: Closeable) => {
-      source.close()
-
-      const { mesh, name } = params
-
-      return api.getMeshGatewayDataplane({ mesh, name })
+    '/meshes/:mesh/dataplanes/:name/rules': async (params) => {
+      return api.getDataplaneRules(params)
     },
 
-    '/meshes/:mesh/dataplane-overviews/:name': (params: DetailParams, source: Closeable) => {
-      source.close()
-
-      const { mesh, name } = params
-
-      return api.getDataplaneOverviewFromMesh({ mesh, name })
+    '/meshes/:mesh/dataplanes/:name/gateway-dataplane-policies': (params) => {
+      return api.getMeshGatewayDataplane(params)
     },
 
-    '/meshes/:mesh/dataplanes/of/:type': async (params: CollectionParams & DataplaneTypeParams & PaginationParams, source: Closeable) => {
-      source.close()
+    '/meshes/:mesh/dataplane-overviews/:name': async (params) => {
+      return api.getDataplaneOverviewFromMesh(params)
+    },
 
-      const { mesh, size, type } = params
+    '/meshes/:mesh/dataplanes/of/:type': async (params) => {
+      const { mesh, size } = params
       const offset = size * (params.page - 1)
 
       const filterParams = Object.fromEntries(normalizeFilterFields(JSON.parse(params.search || '[]')))
 
-      const gatewayParams: DataPlaneOverviewParameters = {}
-      if (type === 'standard') {
-        gatewayParams.gateway = 'false'
-      } else if (type !== 'all') {
-        gatewayParams.gateway = type
-      }
+      const type = params.type === 'standard' ? 'false' : params.type
+      const gatewayParams = includes(['delegated', 'builtin', 'false'] as const, type)
+        ? { gateway: type }
+        : {}
```

Comment:
**Question**: Why does the implementation of this change? Is this a rebase mistake?